adapters/repos/db/index.go

```diff
@@ -1287,19 +1288,23 @@ func (i *Index) multiObjectByID(ctx context.Context,
 		var objects []*storobj.Object
 		var err error
 
+		// TODO follow up task for 1.33 to have searches optionally use CL "for real"?
```

Comment:
The "search" changes here ("search" meaning: "does not use the replicator and uses `i.remote` instead") usually follow the pattern of:
- If the shard is nil or is a replica movement is in progress, use the `remote` method. Note that the `remote` method may still write to localhost, but this is convenient because it allows us to use the same code path during the replica movement while automatically cutting over to writing to the new replica at the right time (if it is a move operation). The downside is there may be some performance cost to using `remote` when we could have just used the `shard` method directly.
- Otherwise, just use the `shard` method directly.

Reply:
Today I'm going to try @salvatore-campagna-weaviate 's suggestion of trying to use the router directly, will probably create a new branch for that work so i can keep them separate